app/controllers/companies_controller.rb

```diff
@@ -25,4 +24,13 @@ def update_companies_details
     flash[:notice] = 'Companies details successfully updated'
     redirect_to companies_url
   end
+
+  def validate_params
+    validator = ParamsValidator.new(params.permit(:query, :stock, :condition))
+
+    return if validator.valid?
+
+    flash[:error] = 'Unknown search params.'
+    redirect_to companies_url
```

Comment:
I think the proper behaviour would be to render the page with error directly instead of adding redirect into the flow.

Reply:
Ok, I thought that it is good enough with message only, but I see now that with error page it is proper behaviour